packages/file-selector/src/lib/useUploadCore.tsx

```diff
@@ -20,12 +24,13 @@ async function startUpload(file: File, options: Options) {
   return upload;
 }
 
-export function useUploadCore(file: File, options: Options) {
+export function useUploadCore(file: File, options: Options, queryOptions?: UploadQueryOptions) {
   const isQueryEnabled = !!file;
 
   return useQuery(['upload', file.name, options], () => startUpload(file, options), {
     enabled: isQueryEnabled,
     retry: false,
     refetchOnWindowFocus: false,
+    ...queryOptions,
   });
```

Comment:
Interesting using useQuery to perform a mutation, but this is the deprecated way of using useQuery. I suggest making this change since it is already being touched.
```suggestion
  return useQuery({
    queryKey: ['upload', file.name, options],
    queryFn: () => startUpload(file, options),
    enabled: isQueryEnabled,
    retry: false,
    refetchOnWindowFocus: false,
    ...queryOptions,
});
```

Reply:
I chose `useQuery` mainly for the automatic calling and state management. Good point updating to the new syntax